### PR TITLE
refactor!: change model to expect mel-band oriented tensors instead of time-oriented ones

### PR DESCRIPTION
**Pull request by roedoejet**
See https://github.com/EveryVoiceTTS/EveryVoice/pull/572

### CODE REVIEW

**Review by SamuelLarkin — CHANGES_REQUESTED** [review]
make the code a little bit more readable

File: hfgl/cli.py [hunk 1]
```diff
@@ -132,28 +132,51 @@ def synthesize(
...
+        raise ValueError(
+            f"Your model expects a spectrogram of dimensions [K (Mel bands), T (frames)] where K == {checkpoint['hyper_parameters']['config']['preprocessing']['audio']['n_mels']} but you provided a tensor of size {data_size}"
+        )
+    if (
+        data_size[0]
+        != checkpoint["hyper_parameters"]["config"]["preprocessing"]["audio"]["n_mels"]
+    ):
+        raise ValueError(
+            f"We expected the first dimension of your Mel spectrogram to correspond with the number of Mel bands declared by your model ({checkpoint['hyper_parameters']['config']['preprocessing']['audio']['n_mels']}). Instead, we found you model has the dimensions {data_size}. If your spectrogram is time-oriented, please re-run this command with the '--time-oriented' flag."
+        )
```
[the remark this answers is not in the document]
Comment (joanise) [reply hfgl/cli.py:169]: Yes, please accept this suggestion, the code is quite hard to read.

File: hfgl/cli.py [hunk 2]
```diff
@@ -132,28 +132,51 @@ def synthesize(
...
+    ):
+        raise ValueError(
+            f"Your model expects a spectrogram of dimensions [K (Mel bands), T (frames)] where K == {checkpoint['hyper_parameters']['config']['preprocessing']['audio']['n_mels']} but you provided a tensor of size {data_size}"
+        )
+    if (
+        data_size[0]
+        != checkpoint["hyper_parameters"]["config"]["preprocessing"]["audio"]["n_mels"]
+    ):
+        raise ValueError(
+            f"We expected the first dimension of your Mel spectrogram to correspond with the number of Mel bands declared by your model ({checkpoint['hyper_parameters']['config']['preprocessing']['audio']['n_mels']}). Instead, we found you model has the dimensions {data_size}. If your spectrogram is time-oriented, please re-run this command with the '--time-oriented' flag."
```
Comment (joanise) [inline hfgl/cli.py:168]: This warning message is good, probably giving the user the feedback I was asking for in the top-level PR.

**Review by joanise — CHANGES_REQUESTED** [review]
Please accept Sam's change request, otherwise this looks fine.

**Comment by joanise** [comment]
Note: you'll need to rebase, and solve the merge conflict: we now have to use `shell_complete`, not `autocompletion`.

**Pull request merged**